README.md

```diff
@@ -51,10 +51,138 @@ pip install -r requirements.txt
 
 
 ## Setup Engine
-- Place your engine(s) in the `engine.dir` directory
-- In `config.yml`, enter the binary name as the `engine.name` field (In Windows you may need to type a name with ".exe", like "lczero.exe")
+- Place your engine(s) in the `engine: dir` directory
+- In `config.yml`, enter the binary name as the `engine: name` field (In Windows you may need to type a name with ".exe", like "lczero.exe")
 - Leave the `weights` field empty or see LeelaChessZero section for Neural Nets
 
+### Engine Configuration
+Besides the above, there are many possible options for configuring the engine for use with lichess-bot.
+
+- `protocol`: Specify which protocol your engine uses. Choices are
+    1. `"uci"` for the [Universal Chess Interface](http://wbec-ridderkerk.nl/html/UCIProtocol.html)
+    2. `"xboard"` for the XBoard/WinBoard/[Chess Engine Communication Protocol](https://www.gnu.org/software/xboard/engine-intf.html)
+    3. `"homemade"` if you want to write your own engine in Python within lichess-bot. See **Creating a custom bot** below.
+- `polyglot`: Tell lichess-bot whether your bot should use an opening book. Multiple books can be specified for each chess variant.
+    - `enabled`: Whether to use the book at all.
+    - `book`: A nested list of books. The next indented line should list a chess variant (`standard`, `3check`, `horde`, etc.) followed on succeeding indented lines with paths to the book files. See `config.yml.default` for examples.
+- `engine_options`: Command line options to pass to the engine on startup. For example, the `config.yml.default` has the configuration
+```
```

Comment:
Over here as well as in lines 105, 113, 128, 137, 164, 172 and 181 of your branch could your change it to something like, “```yml”